src/app/pages/connect-page/_services/connect.api.service.ts

```diff
@@ -40,4 +40,23 @@ export class ConnectApiService {
             });
         })
     }
+
+    removeConnection(requestingUserId: number, connectedWithUserId: number) {
+        const url = environment.apiUrl + '/api/connect';
+        const connectionRemovalRequest = {
+          requestingUserId: requestingUserId,
+          connectedWithUserId: connectedWithUserId
+        };
+        return new Promise((resolve, reject) => {
+            this._apiService.delete(url, connectionRemovalRequest).subscribe(
```

Comment:
apiService.delete() already returns a promise.. you can just return that.. no need to wrap it.

* Side thought.. we are probably doing this in a bunch of places, unnecessarily wrapping a promise.. we should fix that.